cdci_data_analysis/analysis/drupal_helper.py

```diff
@@ -52,39 +58,90 @@ def execute_drupal_request(url,
-                                headers=headers)
-        elif method == 'post':
-            if data is None:
-                data = {}
-            if params is None:
-                params = {}
-            params['_format'] = request_format
-            return requests.post(url,
-                                 params={**params},
-                                 data=data,
-                                 files=files,
-                                 headers=headers
-                                 )
-    except Exception as e:
-        logger.warning(f"an issue occurred when performing a request to the product gallery, "
-                       f"this is likely to be a connection related problem, we are investigating and "
-                       f"try to solve it as soon as possible")
-        if sentry_client is not None:
-            sentry_client.capture('raven.events.Message',
-                                  message=f'exception when performing a request to the product gallery: {repr(e)}')
-        else:
-            logger.warning("sentry not used")
-
-        raise InternalError('issue when performing a request to the product gallery',
-                         status_code=500,
-                         payload={'error_message': str(e)})
+    n_tries_left = n_max_tries
+    global total_n_successful_post_requests, total_n_post_request_retries
+    while True:
+        try:
+            if method == 'get':
+                if params is None:
+                    params = {}
+                params['_format'] = request_format
+                res = requests.get(url,
+                                   params={**params},
+                                   headers=headers)
+
+            elif method == 'post':
+                if data is None:
+                    data = {}
+                if params is None:
+                    params = {}
+                params['_format'] = request_format
+                res = requests.post(url,
+                                    params={**params},
+                                    data=data,
+                                    files=files,
+                                    headers=headers
+                                    )
+            else:
+                raise NotImplementedError
+            if res.status_code == 403:
+                try:
+                    response_json = res.json()
+                    # a 403 has been noticed to be returned in two different cases:
+                    # * for not-valid token
+                    # * not-completed request
+                    error_msg = response_json['message']
+                except json.decoder.JSONDecodeError:
+                    error_msg = res.text
+                raise RequestNotAuthorized(error_msg)
+
+            elif res.status_code not in [200, 201]:
+                logger.warning(f"there seems to be some problem in completing a request to the product gallery:\n"
```

Comment:
I get this:

```
the requested url http://cdciweb02.isdc.unige.ch/mmoda/gallery/observations/range/2018-31-12T23:59:59--2021-01-12T00:00:01 lead to the error {"message":"No route found for \u0022GET \/observations\/range\/2018-31-12T23:59:59--2021-01-12T00:00:01\u0022"}, this might be due to an error in the url or the page requested no longer exists, please check it and try to issue again the request
```

did it pass for you before the push?

Reply:
I am still investigating, I suspect some configuration within the view for the observation, I am seeing the very same error

Reply:
ok it works for me now, for some reasons I had disabled the view :( I must have been struggling too much on these formats

Reply:
works for me too.